styx-client/src/main/java/com/spotify/styx/client/auth/GoogleIdTokenAuth.java

```diff
@@ -33,21 +33,30 @@
 import com.google.api.client.json.webtoken.JsonWebToken.Payload;
 import java.io.IOException;
 import java.security.GeneralSecurityException;
+import java.util.Optional;
 
 public class GoogleIdTokenAuth {
-  private final HttpTransport httpTransport;
   private static final JsonFactory JSON_FACTORY = Utils.getDefaultJsonFactory();
 
-  GoogleIdTokenAuth(HttpTransport httpTransport) {
+  private final HttpTransport httpTransport;
+  private final Optional<GoogleCredential> credential;
+
+  GoogleIdTokenAuth(
+      HttpTransport httpTransport,
+      Optional<GoogleCredential> credential) {
     this.httpTransport = httpTransport;
+    this.credential = credential;
   }
 
-  public GoogleIdTokenAuth() {
-    this(Utils.getDefaultTransport());
+  public Optional<String> getToken(String targetAudience)
+      throws IOException, GeneralSecurityException {
+    return credential.isPresent()
```

Comment:
can be simplified with a `map` I think.

Reply:
`Optional.map()` does not compose well with checked exceptions =/

Reply:
oh, missed that again. sorry. have being programming too much Kotlin these days.